frontend/src/features/leases/detail/LeasePages/insurance/InsuranceContainer.tsx

```diff
@@ -0,0 +1,51 @@
+import { INSURANCE_TYPES } from 'constants/API';
+import { getIn, useFormikContext } from 'formik';
+import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
+import { IInsurance, ILease } from 'interfaces';
+import { useState } from 'react';
+import { Button } from 'react-bootstrap';
+
+import InsuranceDetailsView from './details/Insurance';
+import InsuranceEditContainer from './edit/EditContainer';
+
+const InsuranceContainer: React.FunctionComponent = () => {
+  const context = useFormikContext<ILease>();
+  const leaseId: number = getIn(context.values, 'id') || -1;
+  const insuranceList: IInsurance[] = getIn(context.values, 'insurances') ?? [];
+
+  const lookupCodes = useLookupCodeHelpers();
+  const insuranceTypes = lookupCodes.getByType(INSURANCE_TYPES).sort((a, b) => {
+    return (a.displayOrder || 0) - (b.displayOrder || 0);
+  });
+
+  const [isEditing, setEditing] = useState(false);
+  return (
+    <>
+      {!isEditing && (
```

Comment:
this needs to be controlled by the lease edit claim. So
const { hasClaim } = useKeycloakWrapper();
and then hasClaim(Claims.LEASE_EDIT)

Reply:
To add to this - I thought we agreed to have separate "view" and "edit/create/delete" forms to avoid a single entry point like this container that either shows read-only form or edit form

Having separate "containers" for edit vs details would also allow to use proper claims on each one and each would be on their own FE route: (/leases/id/insurance vs leases/id/insurance/edit)

It would also solve the need for that "hacky" triggerRefresh - once you finish with the edit form you just redirect to the read route which would re-fetch the updated info from the server - or am I missing something?

Reply:
I like it. For now I rather leave it as is since that refactoring looks like is part of a bigger conversation about the sub-view organization.